src/app/app.component.ts

```diff
@@ -1,9 +1,36 @@
 import { Component } from '@angular/core';
-
+import { Observable } from 'rxjs/Observable';
+import { CounterActions } from './actions/counterActions'
+import { Store } from '@ngrx/store';
 @Component({
   selector : 'my-app',
-  template: '<div>Hello App</div>'
+  template: `
+		<button (click)="increment()">Increment</button>
+		<div>Current Count: {{ counter | async}}</div>
+		<button (click)="decrement()">Decrement</button>
+		<button (click)="reset()">Reset Counter</button>
+	`
 })
+
 export class AppComponent {
+  counter: Observable<number>;
+
+	constructor(
+    private store: Store<any>,
+    private counterActions: CounterActions
+  ) {
+		this.counter = store.select('counter');
+	}
+
+	increment() {
+		this.counterActions.increment();
+	}
+
+	decrement() {
```

Comment:
just a question. Can we write dispatch event here(if it's just UI action than we can directly do it here I guess  )? I think you did this here for building framework .

```
decrement() {
		this.store.dispatch({ type: DECREMENT });
	}```